File: src/main/java/com/github/schlak/database/Implementation/MSSQL/FixedValues/MSSQLQueryTypes.java
package com.github.schlak.database.Implementation.MSSQL.FixedValues;


import com.github.schlak.database.Definition.FixedValues.QueryTypes;

/**
 * Created by jonas on 15.03.17.
 */
public enum MSSQLQueryTypes {

    CREATE, UPDATE, DELETE, INSERT;


    public static MSSQLQueryTypes get(QueryTypes queryTypes) {
        if (queryTypes == null) throw new NullPointerException();
        switch (queryTypes) {
            case CREATE:
                return CREATE;
            case UPDATE:
                return UPDATE;
            case DELETE:
                return DELETE;
            case INSERT:
                return INSERT;
        }
        return null;
    }

}

File: src/main/java/com/github/schlak/database/Implementation/MSSQL/FixedValues/MSSQLQueryTypes.kt
package com.github.schlak.database.Implementation.MSSQL.FixedValues

import com.github.schlak.database.Definition.FixedValues.BasicQueryTypes

/**
 * Created by Jonas Schlak on 21.03.2017.
 */

enum class MSSQLQueryTypes {

    CREATE,
    UPDATE,
    DELETE,
    INSERT;

    companion object {
        fun get(basicQueryType: BasicQueryTypes): MSSQLQueryTypes {
            when (basicQueryType) {
                BasicQueryTypes.CREATE -> return CREATE;
                BasicQueryTypes.DELETE -> return DELETE;
                BasicQueryTypes.INSERT -> return INSERT;
                BasicQueryTypes.UPDATE -> return UPDATE;
            }
        }
    }

}